src/main/java/com/iexec/core/replicate/ReplicatesHelper.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.core.replicate;
+
+import com.iexec.common.replicate.ReplicateStatus;
+import com.iexec.common.replicate.ReplicateStatusUpdate;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.iexec.common.replicate.ReplicateStatus.*;
+
+public class ReplicatesHelper {
+    private ReplicatesHelper() {
+
+    }
+
+    /**
+     * Computes the number of replicates in the {@link ReplicateStatus#CONTRIBUTED} status
+     * that have the right contribution hash.
+     * <p>
+     * Note this method won't retrieve the replicates list, so it could be static.
+     * For test purposes - i.e. mocking -, it is not.
+     *
+     * @param replicates       List of replicates on which the calculation should be base.
+     * @param contributionHash Valid hash of the final result.
+     * @return Number of winners who had contributed with the right hash.
+     */
+    public static int getNbValidContributedWinners(List<Replicate> replicates, String contributionHash) {
```

Comment:
Why not putting these methods (methods with `List<Replicate>` as argument) in `ReplicateList`?

Reply:
Sure!